docs/game_coordinator.md

```diff
@@ -0,0 +1,62 @@
+# Game Coordinator
+
+To allow two players to play together, OpenTTD uses a Game Coordinator to
+facilitate this.
+
+When a server starts, it can register itself to the Game Coordinator. This
+happens when `server_game_type` is set to either `invite-only` or `public`.
+Upon registration, the Game Coordinator probes the network of the server, and
+assigns the server an unique code, called an `invite code`.
+
+When a client wants to join a server, it asks the Game Coordinator to help
+with this by giving it the `invite code` of the server. The Game Coordinator
+will, in this order, attempt several ways to connect the client and server
+together:
+
+## 1) Via direct IPv4 / IPv6
+
+Upon registration, the Game Coordinator probed the server to see if based on
+his public IPv4 and/or IPv6, it allows other people from the Internet to
+connect to the game port. If successful, a client is first asked to try this.
```

Comment:
The whole sentence is a bit (over?) complicated. Maybe:
Just after registration the Game Coordinator probes the server, using its public IP address, to see if it allows players to connect directly. If successful, clients are asked to try this method first.